ext/nnstreamer/tensor_filter/tensor_filter_llama2.cc

```diff
@@ -149,13 +149,17 @@ TensorFilterLlama2c::invoke (const GstTensorMemory *input, GstTensorMemory *outp
   char *result = NULL, *prompt, *output_data;
   int len, string_idx = 0, i;
 
-  prompt = (char *) input[0].data;
+  prompt = (char *) malloc (input[0].size + 1);
+  memcpy (prompt, input[0].data, input[0].size);
+  prompt[input[0].size] = '\0';
+
   len = strlen (prompt);
 
-  if (len > 1) {
-    prompt[strlen (prompt) - 1] = ' '; // Remove new line from text file
-  } else {
-    prompt = NULL;
+  if (len > 0) {
+    /* Remove new line from text file */
+    char c = prompt[len - 1];
```

Comment:
I may have misunderstood the code. If we modify it like this, will it behave incorrectly?
```
  prompt = (char *) input[0].data;
  len = strlen (prompt);

  if (len > 0) {
    char c = prompt[len - 1];
    if (c == '\r' || c == '\n')
      prompt[len - 1] = '\0';
  }
```

Reply:
If input[0].data is not end with `\r` or `\n` and if there are some data right behind input[0].data, llama2_api could not know where the string ends.
I added memcpy because I thought it would be better to fix the sub plugin than fix the API.

Reply:
Thank you for the detailed explanation. :pray: